## DualMomentum 
**<div style="text-align: right"> author : 정호영 </div>**

### 배경
모멘텀 현상은 행동재무학에서 일컫는 군집 행동, 정박 효과, 확증 편향, 처분 효과 등의 행동 편향에 의해 발생한다고 알려져 있다.  
- 군집 행동 : 다수 그룹의 행동을 따라하는 경향  
- 정박 효과 : 정보를 처음 제공받은 시점에 지나치게 의존하는 경향  
- 확증 편향 : 본인의 믿음과 반대되는 정보를 무시하는 경향  
- 처분 효과 : 수익이 난 주식을 금방 팔고, 손해 봅 주식을 계속 보유하는 경향  

위의 4개 편향들 살펴보면 다들 각자 위의 경험이 다들 있을 것일텐데, 이를 활용해서 돈을 잘먹고 나오는게 중요하다고 생각합니다.  

### 간단 개념 
- 모멘텀
: "특정 날짜 전"의 주식 가격에 비해 "현재 날짜"의 주식 가격이 얼마나 증가 혹은 감소했는지 알려주는 지표  


- ex) 어제보다 오늘 코스피가 2% 상승 ->, "아, 코스피 하루 모멘텀이 2%네"  
1년전에 비해 20%가 올랐다면 "코스피 1년 모멘텀이 20%"  


- 상대 모멘텀  
특정 기간 동안 상대적으로 수익률이 좋았던 n개 종목을 구함  


- 절대 모멘텀  
자산의 가치가 상승하고 있을 때만 투자하고, 그렇지 않을 떄는 단기 국채를 매수하거나 현금을 보유하는 전략  


- 모멘텀 전략?
현재주가와 과거 주가를 비교해서 현재주가가 높으면 매수, 낮으면 매도  


- 약점  
횡보장에서 구조적으로 손실이 누적됨. 일시적인 급락과 급등이 연이어 이어지는 시장이면서, 큰 추세없이 변동성만 큰 시장이 되면 모멘텀 전략이 잘 작동하지 않는다고 함

### 모멘텀을 조금 더 deep하게 참고하려면  
http://blog.naver.com/PostView.nhn?blogId=htk1019&logNo=221070300022

### Application
개인적으로 관심이 있는 미국 주식 종목들 대상으로 테스트해볼 예정

In [1]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
from slacker import Slacker
import time
from pandas_datareader import data as pdr 
import yfinance as yf
yf.pdr_override()
import pymysql

In [4]:
# 하나의 종목만 보고 싶을 때
tsla = pdr.get_data_yahoo(tickers="TSLA",
                          start = '2020-01-01',
                          end = '2020-11-07',
                         auto_adjust=True,
                         )
tsla.head()

[*********************100%***********************]  1 of 1 completed


Open       High        Low      Close    Volume
Date                                                            
2019-12-31  81.000000  84.258003  80.416000  83.666000  51428500
2020-01-02  84.900002  86.139999  84.342003  86.052002  47660500
2020-01-03  88.099998  90.800003  87.384003  88.601997  88892500
2020-01-06  88.094002  90.311996  88.000000  90.307999  50665000
2020-01-07  92.279999  94.325996  90.671997  93.811996  89410500

In [6]:
!pip -q install finance-datareader

In [7]:
import FinanceDataReader as fdr
# https://github.com/FinanceData/FinanceDataReader
# https://nbviewer.jupyter.org/github/FinanceData/FinanceDataReader/blob/master/tutorial/FinanceDataReader%20Tutorial%20-%20%EC%84%B9%ED%84%B0%20%ED%8F%89%EA%B7%A0%20%EC%88%98%EC%9D%B5%EB%A5%A0%EA%B3%BC%20%EA%B0%9C%EB%B3%84%20%EC%A2%85%EB%AA%A9%EC%9D%98%20%EC%88%98%EC%9D%B5%EB%A5%A0.ipynb

In [13]:
nasdaq = fdr.StockListing('NASDAQ') 
nasdaq.head()
# # NYSE, NASDAQ, AMEX stock symbol list
# stocks = fdr.StockListing('NYSE')   # 뉴욕거래소
# stocks = fdr.StockListing('NASDAQ') # 나스닥
# stocks = fdr.StockListing('AMEX')   # 아멕스

# # S&P 500 symbol list
# sp500 = fdr.StockListing('S&P500')

100%|█████████████████████████████████████████████████████████████████████████████| 3162/3162 [00:04<00:00, 389.48it/s]
C:\Users\user\Anaconda3\lib\site-packages\FinanceDataReader\naver\listing.py:85: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  merged = pd.concat(df_list)


Symbol                  Name        Industry IndustryCode
0   AAPL             Apple Inc  컴퓨터, 전화 및 가전제품       571060
1   MSFT        Microsoft Corp   소프트웨어 및 IT서비스       572010
2   AMZN        Amazon.com Inc         다양한 소매업       534020
3     FB  Facebook Inc Class A   소프트웨어 및 IT서비스       572010
4  GOOGL  Alphabet Inc Class A   소프트웨어 및 IT서비스       572010

In [16]:
# count, mean, sum 적용해서 살펴 보세요~
# nasdaq.groupby('Industry').count()

In [15]:
nasdaq.groupby('Industry').count().sort_values('Symbol', ascending=False)[:30]

Symbol  Name  IndustryCode
Industry                                     
생명과학 및 메디컬 리서치        532   532           532
은행                    393   393           393
소프트웨어 및 IT서비스         302   302           302
기타 금융업                234   234           234
헬스케어 장비 및 용품          169   169           169
제약                    132   132           132
전문 및 상업 서비스            95    95            95
반도체 및 반도체 장비           90    90            90
건설장비 및 대형차, 기차, 배      78    78            78
미디어 및 출판               75    75            75
호텔 및 엔터테인먼트            66    66            66
보험                     64    64            64
투자은행 및 서비스             59    59            59
커뮤니케이션 및 네트워킹          54    54            54
주택 및 상업용 리츠            54    54            54
식품 및 담배                53    53            53
다양한 금융서비스              47    47            47
전문품 상점                 44    44            44
화물 및 물류 서비스            44    44            44
헬스케어 업체 및 서비스          43    43            43
오일 및 가스 장비 및 서비스       34    34            34
컴퓨터, 전화 및 가전제품         31    31            31
통신 서비스                 30    30            30
전자 장비 및 부품             30    30            30
부동산 운영 관리              29    29            29
자동차 및 자동차 부품           28    28            28
재생에너지                  27    27            27
오일 및 가스                24    24            24
주택건설                   22    22            22
레저 상품                  22    22            22

묶은(groupby)결과를 list와 dict로 변환하면 개별 섹터(Sector)별로 종목리스트를 구분할 수 있습니다.

In [17]:
sectors = dict(list(nasdaq.groupby('Industry')))

print('count:', len(sectors))
list(sectors.keys())[:10]

count: 57


['가정용품',
 '개인, 가정 용품 및 서비스',
 '건설 및 엔지니어링',
 '건설장비 및 대형차, 기차, 배',
 '건축 자재',
 '교육서비스',
 '기타 금융업',
 '다양한 금융서비스',
 '다양한 산업재 도매업',
 '다양한 소매업']

In [19]:
it_sec = sectors['소프트웨어 및 IT서비스']

print('row count:', len(it_sec))
it_sec.head(10)

row count: 302


Symbol                                   Name       Industry IndustryCode
1    MSFT                         Microsoft Corp  소프트웨어 및 IT서비스       572010
3      FB                   Facebook Inc Class A  소프트웨어 및 IT서비스       572010
4   GOOGL                   Alphabet Inc Class A  소프트웨어 및 IT서비스       572010
5    GOOG                   Alphabet Inc Class C  소프트웨어 및 IT서비스       572010
8    PYPL                    PayPal Holdings Inc  소프트웨어 및 IT서비스       572010
9    ADBE                              Adobe Inc  소프트웨어 및 IT서비스       572010
10   NFLX                            Netflix Inc  소프트웨어 및 IT서비스       572010
23     ZM  Zoom Video Communications Inc Class A  소프트웨어 및 IT서비스       572010
24    PDD                      Pinduoduo Inc ADR  소프트웨어 및 IT서비스       572010
30   INTU                             Intuit Inc  소프트웨어 및 IT서비스       572010

## 개별 종목의 가격 데이터 가져오기
302개 다들고 오는게 조금 많으니까 적당히 들고와보겠습니다.

In [21]:
it_sec = it_sec.iloc[0:20]

In [61]:
it = pd.DataFrame()

for ix, row in it_sec.iterrows():
    code, name = row['Symbol'], row['Name']
    print(code, name)
    # 개별 종목의 가격을 가져옵니다
    df = fdr.DataReader(code, '2020-01-01', '2020-11-07')

    # 가격 데이터의 종가(Close)를 컬럼으로 추가합니다
    # (컬럼명은 종목명을 지정합니다)
    it[name] = df['Close']

MSFT Microsoft Corp
FB Facebook Inc Class A
GOOGL Alphabet Inc Class A
GOOG Alphabet Inc Class C
PYPL PayPal Holdings Inc
ADBE Adobe Inc
NFLX Netflix Inc
ZM Zoom Video Communications Inc Class A
PDD Pinduoduo Inc ADR
INTU Intuit Inc
MELI Mercadolibre Inc
ADP Automatic Data Processing Inc
NTES NetEase Inc ADR
ATVI Activision Blizzard Inc
ADSK Autodesk Inc
WDAY Workday Inc Class A
TEAM Atlassian Corporation PLC Class A
BIDU Baidu Inc ADR
DOCU DocuSign Inc
CTSH Cognizant Technology Solutions Corp Class A


In [62]:
# 결측치 있는 column 잠시 지우겠습니다
it = it.dropna(axis=1)
print(it.shape)
it

(216, 20)


Microsoft Corp  Facebook Inc Class A  Alphabet Inc Class A  \
Date                                                                     
2020-01-02          160.62                209.78               1368.68   
2020-01-03          158.62                208.67               1361.52   
2020-01-06          159.03                212.60               1397.81   
2020-01-07          157.58                213.06               1395.11   
2020-01-08          160.09                215.22               1405.04   
2020-01-09          162.09                218.30               1419.79   
2020-01-10          161.34                218.06               1428.96   
2020-01-13          163.28                221.91               1440.03   
2020-01-14          162.13                219.06               1430.59   
2020-01-15          163.18                221.15               1439.20   
2020-01-16          166.17                221.77               1450.16   
2020-01-17          167.10                222.14               1479.52   
2020-01-21          166.50                221.44               1482.25   
2020-01-22          165.70                221.32               1484.17   
2020-01-23          166.72                219.76               1484.69   
2020-01-24          165.04                217.94               1466.17   
2020-01-27          162.28                214.87               1431.73   
2020-01-28          165.46                217.79               1450.50   
2020-01-29          168.04                223.23               1456.70   
2020-01-30          172.78                209.53               1454.25   
2020-01-31          170.23                201.91               1432.78   
2020-02-03          174.38                204.19               1482.60   
2020-02-04          180.12                209.83               1445.41   
2020-02-05          179.90                210.11               1446.05   
2020-02-06          183.63                210.85               1475.97   
2020-02-07          183.89                212.33               1479.11   
2020-02-10          188.70                213.06               1508.66   
2020-02-11          184.44                207.19               1510.06   
2020-02-12          184.71                210.76               1518.63   
2020-02-13          183.71                213.14               1513.39   
...                    ...                   ...                   ...   
2020-09-28          209.44                256.82               1458.66   
2020-09-29          207.26                261.79               1466.02   
2020-09-30          210.33                261.90               1465.60   
2020-10-01          212.46                266.63               1487.90   
2020-10-02          206.19                259.94               1455.60   
2020-10-05          210.38                264.65               1482.83   
2020-10-06          205.91                258.66               1451.02   
2020-10-07          209.83                258.12               1459.14   
2020-10-08          210.58                263.76               1483.43   
2020-10-09          215.81                264.45               1510.45   
2020-10-12          221.40                275.75               1564.59   
2020-10-13          222.86                276.14               1567.07   
2020-10-14          220.86                271.82               1563.44   
2020-10-15          219.66                266.72               1555.47   
2020-10-16          219.66                265.93               1567.70   
2020-10-19          214.22                261.40               1529.95   
2020-10-20          214.65                267.56               1551.08   
2020-10-21          214.80                278.73               1585.99   
2020-10-22          214.89                278.12               1606.66   
2020-10-23          216.23                284.79               1632.98   
2020-10-26          210.08                277.11               1584.29   
2020-10-27      

## 수익률 계산하기

In [63]:
acc_rets = it / it.iloc[0] - 1.0
acc_rets

Microsoft Corp  Facebook Inc Class A  Alphabet Inc Class A  \
Date                                                                     
2020-01-02        0.000000              0.000000              0.000000   
2020-01-03       -0.012452             -0.005291             -0.005231   
2020-01-06       -0.009899              0.013443              0.021283   
2020-01-07       -0.018927              0.015635              0.019311   
2020-01-08       -0.003300              0.025932              0.026566   
2020-01-09        0.009152              0.040614              0.037343   
2020-01-10        0.004483              0.039470              0.044042   
2020-01-13        0.016561              0.057822              0.052131   
2020-01-14        0.009401              0.044237              0.045233   
2020-01-15        0.015938              0.054200              0.051524   
2020-01-16        0.034554              0.057155              0.059532   
2020-01-17        0.040344              0.058919              0.080983   
2020-01-21        0.036608              0.055582              0.082978   
2020-01-22        0.031627              0.055010              0.084381   
2020-01-23        0.037978              0.047574              0.084760   
2020-01-24        0.027518              0.038898              0.071229   
2020-01-27        0.010335              0.024264              0.046066   
2020-01-28        0.030133              0.038183              0.059780   
2020-01-29        0.046196              0.064115              0.064310   
2020-01-30        0.075707             -0.001192              0.062520   
2020-01-31        0.059831             -0.037515              0.046833   
2020-02-03        0.085668             -0.026647              0.083233   
2020-02-04        0.121405              0.000238              0.056061   
2020-02-05        0.120035              0.001573              0.056529   
2020-02-06        0.143257              0.005101              0.078389   
2020-02-07        0.144876              0.012156              0.080684   
2020-02-10        0.174823              0.015635              0.102274   
2020-02-11        0.148300             -0.012346              0.103297   
2020-02-12        0.149981              0.004672              0.109558   
2020-02-13        0.143755              0.016017              0.105730   
...                    ...                   ...                   ...   
2020-09-28        0.303947              0.224235              0.065742   
2020-09-29        0.290375              0.247926              0.071120   
2020-09-30        0.309488              0.248451              0.070813   
2020-10-01        0.322749              0.270998              0.087106   
2020-10-02        0.283713              0.239108              0.063506   
2020-10-05        0.309800              0.261560              0.083402   
2020-10-06        0.281970              0.233006              0.060160   
2020-10-07        0.306375              0.230432              0.066093   
2020-10-08        0.311045              0.257317              0.083840   
2020-10-09        0.343606              0.260606              0.103582   
2020-10-12        0.378409              0.314472              0.143138   
2020-10-13        0.387498              0.316331              0.144950   
2020-10-14        0.375047              0.295738              0.142298   
2020-10-15        0.367576              0.271427              0.136475   
2020-10-16        0.367576              0.267661              0.145410   
2020-10-19        0.333707              0.246067              0.117829   
2020-10-20        0.336384              0.275431              0.133267   
2020-10-21        0.337318              0.328678              0.158773   
2020-10-22        0.337878              0.325770              0.173876   
2020-10-23        0.346221              0.357565              0.193106   
2020-10-26        0.307932              0.320955              0.157531   
2020-10-27      

수익률 반환

In [64]:
returns = acc_rets.iloc[-1]
returns

Microsoft Corp                                 0.392853
Facebook Inc Class A                           0.398656
Alphabet Inc Class A                           0.285713
Alphabet Inc Class C                           0.288422
PayPal Holdings Inc                            0.830519
Adobe Inc                                      0.479024
Netflix Inc                                    0.560686
Zoom Video Communications Inc Class A          6.277503
Pinduoduo Inc ADR                              1.777832
Intuit Inc                                     0.323490
Mercadolibre Inc                               1.435078
Automatic Data Processing Inc                 -0.024014
NetEase Inc ADR                                0.400061
Activision Blizzard Inc                        0.343734
Autodesk Inc                                   0.381089
Workday Inc Class A                            0.343604
Atlassian Corporation PLC Class A              0.612793
Baidu Inc ADR                                  0

수익률 내림차순

In [65]:
returns.sort_values(ascending=False) # zoom 수익률이 627%

Zoom Video Communications Inc Class A          6.277503
DocuSign Inc                                   2.052569
Pinduoduo Inc ADR                              1.777832
Mercadolibre Inc                               1.435078
PayPal Holdings Inc                            0.830519
Atlassian Corporation PLC Class A              0.612793
Netflix Inc                                    0.560686
Adobe Inc                                      0.479024
NetEase Inc ADR                                0.400061
Facebook Inc Class A                           0.398656
Microsoft Corp                                 0.392853
Autodesk Inc                                   0.381089
Activision Blizzard Inc                        0.343734
Workday Inc Class A                            0.343604
Intuit Inc                                     0.323490
Alphabet Inc Class C                           0.288422
Alphabet Inc Class A                           0.285713
Cognizant Technology Solutions Corp Class A    0

In [66]:
returns.mean() # 기술주 평균 수익률 올해(10개월) 87%

0.8700036674604407

## 다양한 기간에 대한 수익률
기간을 달리해서 수익률 추출

In [95]:
df = it['2020-10-01':'2020-10-31'] # 특정 기간(12월 1달) 동안
acc_rets = df / df.iloc[0] - 1.0
acc_rets.iloc[-1]

Microsoft Corp                                -0.047021
Facebook Inc Class A                          -0.013202
Alphabet Inc Class A                           0.086168
Alphabet Inc Class C                           0.087860
PayPal Holdings Inc                           -0.054890
Adobe Inc                                     -0.104923
Netflix Inc                                   -0.098140
Zoom Video Communications Inc Class A         -0.045715
Pinduoduo Inc ADR                              0.208921
Intuit Inc                                    -0.053138
Mercadolibre Inc                               0.088502
Automatic Data Processing Inc                  0.147131
NetEase Inc ADR                               -0.063956
Activision Blizzard Inc                       -0.084059
Autodesk Inc                                   0.006753
Workday Inc Class A                           -0.047161
Atlassian Corporation PLC Class A              0.030160
Baidu Inc ADR                                  0

'2020-10-31' 시점을 기준으로 과거 5일, 10일, 20일, 60일, 120일, 240일 각각 수익률을 구해봅니다.

각 기간의 (시작날짜, 끝날짜)는 다음과 같이 생성할 수 있습니다.

In [101]:
from datetime import datetime, timedelta

the_day = datetime(2020, 10, 31)

for days in [5, 10, 20, 60, 120, 180, 240]:
    start = the_day - timedelta(days)
    end = the_day
    print(start, '~', end)

2020-10-26 00:00:00 ~ 2020-10-31 00:00:00
2020-10-21 00:00:00 ~ 2020-10-31 00:00:00
2020-10-11 00:00:00 ~ 2020-10-31 00:00:00
2020-09-01 00:00:00 ~ 2020-10-31 00:00:00
2020-07-03 00:00:00 ~ 2020-10-31 00:00:00
2020-05-04 00:00:00 ~ 2020-10-31 00:00:00
2020-03-05 00:00:00 ~ 2020-10-31 00:00:00


In [102]:
from datetime import datetime, timedelta

the_day = datetime(2020, 10, 31)
row_dict = {}
for days in [5, 10, 20, 60, 120, 180, 240]:
    start = the_day - timedelta(days)
    end = the_day

    df = it[start:end] # 특정 기간
    acc_rets = df / df.iloc[0] - 1.0
    row_dict[days] = acc_rets.iloc[-1] 

In [103]:
# day_interval = ['day_' + str(df_rets.columns[i]) for i in range(len(df_rets.columns))]
# df_rets = pd.DataFrame(row_dict,
#                       columns = day_interval)
df_rets

Empty DataFrame
Columns: [day_day_5, day_day_10, day_day_20, day_day_60, day_day_120, day_day_180, day_day_240]
Index: []

In [92]:
pd.DataFrame(row_dict).columns #,
              #columns = day_interval)

Int64Index([5, 10, 20, 60, 120, 180, 240], dtype='int64')

In [44]:
# it 섹터 전체 기간별 수익률
df_rets.mean()

5     -0.028012
10    -0.044339
20    -0.040368
60    -0.059940
120    0.082987
180    0.455065
240    0.520500
dtype: float64

## 데이터프레임 Style 부여

In [45]:
def color_negative_red(val):
    color = 'red' if val < 0 else 'black'
    return 'color: %s' % color

def highlight_max(s):
    is_max = s == s.max()
    return ['background-color: yellow' if v else '' for v in is_max]

df_rets.style.\
    applymap(color_negative_red).\
    apply(highlight_max)

In [ ]:
## 엑셀로 저장하고 싶으면
# df_rets.style.\
#     applymap(color_negative_red).\
#     apply(highlight_max).\
#     to_excel('기간별수익률데이터나스닥(2020.10.31).xlsx', engine='openpyxl')

## 모멘텀을 위해 책에 있는 Tabular 형태로 만들기

In [52]:
# df_rets = df_rets.reset_index()
# df_rets.rename(columns = {'index':'Name'}, inplace = True)
# it = it.transpose().reset_index()
# it.rename(columns = {'index':'Name'}, inplace = True)
# rm = pd.merge(df_rets, it, on = 'Name', how = 'inner')
# rm

Name         5        10        20  \
0                                Microsoft Corp -0.036224 -0.057402 -0.085501   
1                          Facebook Inc Class A -0.050521 -0.056040 -0.045839   
2                          Alphabet Inc Class A  0.020085  0.018991  0.032929   
3                          Alphabet Inc Class C  0.019215  0.017385  0.033050   
4                           PayPal Holdings Inc -0.056232 -0.126437 -0.076324   
5                                     Adobe Inc -0.059133 -0.098516 -0.124861   
6                                   Netflix Inc -0.025602 -0.027216 -0.118690   
7         Zoom Video Communications Inc Class A -0.109851 -0.101873 -0.062314   
8                             Pinduoduo Inc ADR  0.035205 -0.002771  0.062087   
9                                    Intuit Inc -0.040873 -0.054419 -0.103297   
10                             Mercadolibre Inc -0.052589 -0.065490 -0.001398   
11                Automatic Data Processing Inc  0.083625  0.075655  0.049010   
12                              NetEase Inc ADR  0.000923 -0.016432 -0.051475   
13                      Activision Blizzard Inc -0.059138 -0.075787 -0.065754   
14                                 Autodesk Inc -0.054550 -0.072714 -0.027177   
15                          Workday Inc Class A -0.000998 -0.063261 -0.079188   
16            Atlassian Corporation PLC Class A -0.075193 -0.074255 -0.013641   
17                                Baidu Inc ADR  0.000752 -0.013275  0.041732   
18                                 DocuSign Inc -0.081392 -0.072758 -0.127706   
19  Cognizant Technology Solutions Corp Class A -0.017742 -0.020167 -0.043012   

          60       120       180       240  2020-01-02 00:00:00  \
0  -0.109121 -0.039060  0.132129  0.217718               160.62   
1  -0.109430  0.095014  0.281838  0.420911               209.78   
2  -0.023546  0.077658  0.221642  0.229205              1368.68   
3  -0.023905  0.083780  0.221744  0.228932              1367.37   
4  -0.109127  0.056476  0.505175  0.659208               110.75   
5  -0.153140 -0.012130  0.280685  0.273680               334.43   
6  -0.145198 -0.036593  0.111153  0.276195               329.81   
7   0.007035  0.765939  2.213260  2.687280                68.72   
8  -0.015105  0.034253  0.940479  1.491141                41.23   
9  -0.093611  0.016244  0.175144  0.095759               266.16   
10  0.003961  0.219011  0.998765  0.933570               610.19   
11  0.140258  0.047967  0.093300 -0.016867               170.32   
12 -0.117629 -0.013862  0.316596  0.272767                65.74   
13 -0.091204 -0.037371  0.135382  0.230982                58.65   
14 -0.070885 -0.036094  0.324747  0.279830               187.83   
15 -0.130838  0.101316  0.415235  0.271066               167.46   
16 -0.024388  0.006038  0.174646  0.258919               122.88   
17  0.048216  0.000903  0.404073  0.125254               138.22   
18 -0.247582  0.060789  0.888775  1.282731                75.90   
19  0.066448  0.269463  0.266537  0.191724                61.92   

    2020-01-03 00:00:00  ...  2020-10-26 00:00:00  2020-10-27 00:00:00  \
0                158.62  ...               210.08               213.25   
1                208.67  ...               277.11               283.29   
2               1361.52  ...              1584.29              1598.88   
3               1360.66  ...              1590.45              1604.26   
4                108.76  ...               197.22               200.43   
5                331.81  ...               475.20               478.56   
6                325.90  ...               488.24               488.93   
7                 67.28  ...               517.79               538.99   
8                 40.89  ...                86.92                90.31   
9                264.38  ...               328.09               330.27   
10               607.80  ...              1281.44              1282.47   
11               169.96  ...               14